### PR TITLE
Issue78 Move away from Source Code Pro web fonts in favour of system fonts

### PR DESCRIPTION
**Pull request by AJDBenner**
#78

### CODE REVIEW

File: src/extensions/default/InlineColorEditor/css/main.less [hunk 1]
```diff
@@ -457,3 +457,5 @@
         }
     }
 }
+
+@sansFontFamily: Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
```
Comment (Pomax) [inline src/extensions/default/InlineColorEditor/css/main.less]: those are three very WIndows specific fonts, all of which are from the Asian langauges... Where are the regular Sans serif fonts available in modern Windows, and where are the better OSX fonts? (and you probably want those at the front of the family, then, since Helvetica and Arial are considered websafe and will thus always get selected. Any system with the followup fonts has Arial, for instance)

File: src/styles/brackets_fonts.less [hunk 2]
```diff
@@ -27,57 +27,53 @@
 @font-weight-light: 200;
 
 
-/* SourceSansPro Regular */
+/* @sansFontFamily Regular */
```
Comment (Pomax) [inline src/styles/brackets_fonts.less]: curious why these would be kept around if the intention is to no longer use them.

File: src/extensions/default/InlineColorEditor/css/main.less (hunk 1 above)
Comment (sedge) [inline src/extensions/default/InlineColorEditor/css/main.less]: :+1:

Comment (humphd) [inline src/extensions/default/InlineColorEditor/css/main.less]: This is what Brackets does, we're just removing the web font.  I'd rather file a new bug afterward to evaluate our font choices as part of a larger UX overhaul in Thimble vs. doing it in this PR.

File: src/extensions/default/JavaScriptCodeHints/styles/brackets-js-hints.css [hunk 3]
```diff
@@ -45,15 +45,15 @@
 
 .brackets-js-hints.literal-hint {
     color: #444; /* dark grey */
-    font-family: SourceCodePro;
+    font-family: @sourceFontFamily;
```
Comment (humphd) [inline src/extensions/default/JavaScriptCodeHints/styles/brackets-js-hints.css]: This is a `.css` file, not `.less`, so you need to explicitly add these vs. using variables.

File: src/extensions/default/WebPlatformDocs/WebPlatformDocs.less [hunk 4]
```diff
@@ -131,7 +131,7 @@
 }
 
 .divider-holder {
-    // This nesting lets us effectively have 'left: (37% + 12px)'
+    // This nesting lets us effectively have 'left: (37% + 12px)
```
Comment (humphd) [inline src/extensions/default/WebPlatformDocs/WebPlatformDocs.less]: We removed the WebPlatformDocs extension, so you can get rid of the changes to this file altogether.

File: src/styles/brackets_codemirror_override.less [hunk 5]
```diff
@@ -171,7 +171,7 @@ div.CodeMirror-cursors {
 .CodeMirror-linenumber {
     color: @accent-comment;
     min-width: 2.5em;
-    /*font-size: 0.9em;*/  /* restore after SourceCodePro font fix? */
+    /*font-size: 0.9em;*/  /* restore after @sourceFontFamily font fix? */
```
Comment (humphd) [inline src/styles/brackets_codemirror_override.less]: Undo this change so it's more obvious what it refers to

File: src/styles/brackets_codemirror_override.less [hunk 6]
```diff
@@ -272,7 +272,7 @@ div.CodeMirror-cursors {
 
 /*
  * Temporarily override bold and italic syntax highlighting until
- * SourceCodePro supports them in a fixed pitch
+ * @sourceFontFamily supports them in a fixed pitch
```
Comment (humphd) [inline src/styles/brackets_codemirror_override.less]: Same thing here, undo.

File: src/styles/brackets_fonts.less (hunk 2 above)
Comment (humphd) [inline src/styles/brackets_fonts.less]: I don't think we should change any of these `font-face` declarations, since they won't get picked up at all, based on your changes below.  Revert all this, please.

File: src/styles/brackets_fonts.less [hunk 7]
```diff
@@ -27,57 +27,53 @@
...
     font-weight: normal;
     font-style: normal;
 }
 
 /* Font Stacks */
 
-@sansFontFamily: 'SourceSansPro', Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
+@sansFontFamily: Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
+@sourceFontFamily: "Menlo Regular", Consolas, Inconsolata, "Vera Sans", "Lucida Console", Courier, fixed;
+@sourceFontFamily-medium: "Menlo Regular", Consolas, Inconsolata, "Vera Sans", "Lucida Console", Courier, fixed;
```
Comment (humphd) [inline src/styles/brackets_fonts.less]: I'm not sure about this one, it probably needs to happen in another commit on my upstream PR.

File: src/extensions/default/InlineTimingFunctionEditor/main.less [hunk 8]
```diff
@@ -379,3 +379,5 @@
         }        
     }
 }
+
+@sansFontFamily: Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
```
Comment (humphd) [inline src/extensions/default/InlineTimingFunctionEditor/main.less]: Can you confirm that defining variables after they are used is OK with LESS?  I'm not sure, but you might want to move this to the top of the file.  I might be wrong, check it out?

File: src/extensions/default/JavaScriptCodeHints/styles/brackets-js-hints.css [hunk 9]
```diff
@@ -109,7 +109,7 @@
 
 .dark .brackets-js-hints.literal-hint {
     color: #bbb; /* dark grey */
-    font-family: SourceCodePro;
+    font-family: @sourceFontFamily;
```
Comment (humphd) [inline src/extensions/default/JavaScriptCodeHints/styles/brackets-js-hints.css]: No variables in `.css` files, please.  Use a static `font-family` declaration here.

File: src/styles/brackets.less [hunk 10]
```diff
@@ -2167,3 +2167,5 @@ label input {
     min-height: 300px;
     max-height: 300px;
 }
+
+@sourceFontFamily: "Menlo Regular", Consolas, Inconsolata, "Vera Sans", "Lucida Console", Courier, fixed;
```
Comment (humphd) [inline src/styles/brackets.less]: Same question here with regard to order of use/decl.

File: src/styles/brackets_fonts.less [hunk 11]
```diff
@@ -80,4 +80,5 @@
 
 /* Font Stacks */
 
-@sansFontFamily: 'SourceSansPro', Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
+@sansFontFamily: Helvetica, Arial, "Meiryo UI", "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
+@sourceFontFamily: "Menlo Regular", Consolas, Inconsolata, "Vera Sans", "Lucida Console", Courier, fixed;
```
Comment (humphd) [inline src/styles/brackets_fonts.less]: NOTE: I've got a new commit on https://github.com/adobe/brackets/pull/10727 that will do the `-Medium` bit, but it hasn't landed yet.

File: src/styles/brackets.less [hunk 12]
```diff
@@ -41,6 +41,8 @@
  */
 @import "brackets_shared.less";
 
+@sourceFontFamily: "Menlo Regular", Consolas, Inconsolata, "Vera Sans", "Lucida Console", Courier, fixed;
+
```
Comment (humphd) [inline src/styles/brackets.less]: These changes aren't necessary, revert please (they get picked up via LESS import).

File: src/extensions/default/InlineColorEditor/css/main.less (hunk 1 above)
Comment (humphd) [inline src/extensions/default/InlineColorEditor/css/main.less]: Move this up to top, right?
